aten/src/ATen/native/native_functions.yaml

```diff
@@ -4121,6 +4124,9 @@
 - func: softmax.int(Tensor self, int dim, ScalarType? dtype=None) -> Tensor
   variants: function, method
 
+- func: softmax.int_out(Tensor self, int dim, ScalarType? dtype=None, *, Tensor(a!) out) -> Tensor(a!)
+  variants: function
```

Comment:
These out= ops should probably be marked `CompositeExplicitAutograd`, since out= ops aren't supported by autograd (and the "default" in native_functions.yaml is `CompositeImplicitAutograd`, which implies autograd support).

cc @albanD is any other action needed (like explicitly specifying the lack of autograd support)? Or will autograd automatically generate a nice error message when you try to use one of these ops with it.

Reply:
If it is written in a "differentiable" way, then nothing to be done. If you call native ops, then yes you can make it CompositeExplicitAutograd to get a default error kernel.